src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+    }
+
+    /**
+     * {WAITING} -> RUNNING
+     */
+    private void toRUNNING() {
+        assert(state == JobState.WAITING) : "going to RUNNING state from non-WAITING state";
+        assert(thread == null) : "already having job thread while going to RUNNING state";
+        assert(waitingTickerJob != null) : "going to RUNNING state without ticker job";
+        waitingTickerJob = null;
+        nextExecutionTime = NO_EXECUTION;
+        thread = Thread.currentThread();
+        state = JobState.RUNNING;
+    }
+
+    /**
+     * {RUNNING} -> TERMINATING
+     */
+    private void toTERMINATING() {
+        assert(state == JobState.RUNNING) : "going to TERMINATING state from non-RUNNING state";
+        assert(thread != null) : "going to TERMINATING state without job thread";
+        assert(waitingTickerJob == null) : "having ticker job while going to TERMINATING state";
+        thread.interrupt();
+        state = JobState.TERMINATING;
+    }
+
+    /**
+     * {IDLE, TERMINATING, WAITING} -> TERMINATED
+     */
+    private void toTERMINATED() {
+        if (state == JobState.TERMINATING) {
+            assert(thread == Thread.currentThread()) : "going to TERMINATED from non-job thread";
+            assert(waitingTickerJob == null) : "going to TERMINATED with waiting ticker job";
+            thread = null;
+        } else if (state == JobState.WAITING) {
+            assert(thread == null) : "having job thread while going to TERMINATED state";
+            assert(waitingTickerJob != null) : "in WAITING state but no ticker job";
+            // Remove the scheduled job from the ticker on a best-effort basis.
+            ticker.removeQueuedJob(waitingTickerJob);
+            waitingTickerJob = null;
+        } else {
+            assert(state == JobState.IDLE) : "going to TERMINATED state from illegal state";
+            assert(thread == null) : "having job thread while going to TERMINATED state";
+            assert(waitingTickerJob == null) : "going to TERMINATED with waiting ticker job";
+        }
+        state = JobState.TERMINATED;
+        // Notify all threads waiting in waitForTermination()
+        notifyAll();
+    }
+
+    /**
+     * Indicates whether this job may start and sets the state to {@code RUNNING}
+     * accordingly. If this job is {@code TERMINATED}, it may not start, if it is {@code
+     * WAITING}, it may start.
+     * @return {@code true} if the job may start
+     */
+    private synchronized boolean jobStarted() {
+        if (state == JobState.TERMINATED) {
+            return false;
+        }
+        toRUNNING();
+        return true;
+    }
+
+    /**
+     * Finishes the job by either enqueueing itself again in {@code WAITING} (if there has been a
+     * trigger since the start of the last job), waiting for a trigger in {@code IDLE} or going
+     * to the {@code TERMINATED} state if we were previously {@code TERMINATING}.
+     */
+    private synchronized void jobFinished() {
```

Comment:
The name of the function sounds like it is a read-only function, while it actually modifies the state.
I would suggest renaming it to "finishJob()"